subprojects/language-java/src/test/groovy/org/gradle/api/tasks/JavaExecTest.groovy

```diff
@@ -16,16 +16,30 @@
 
 package org.gradle.api.tasks
 
-import spock.lang.Specification
+import org.gradle.api.InvalidUserDataException
+import org.gradle.jvm.toolchain.JavaToolchainService
+import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+import org.gradle.util.TestUtil
 
-class JavaExecTest extends Specification {
+class JavaExecTest extends AbstractProjectBuilderSpec {
 
-    def 'javaLauncher is annotated with @Nested and @Optional'() {
-        given:
-        def launcherMethod = JavaExec.class.getMethod('getJavaLauncher', [] as Class[])
+    def setup() {
+        def toolchainService = Mock(JavaToolchainService)
+        project.extensions.add("javaToolchains", toolchainService)
```

Comment:
Why does this test need the `javaToolchains` extension? I don't think the task should reach into project to get the service.

Reply:
You are right, this can be omitted here.

This will probably be required for the main PR though for some of the unit tests. That is because the task needs to use the `JavaToolchainService` to resolve the executable. In this PR it does not matter, because the exception is thrown right before that.